framework/include/utils/MeshBaseDiagnosticsUtils.h

```diff
@@ -0,0 +1,24 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseTypes.h"
+#include "libmesh/mesh_base.h"
```

Comment:
Can just forward declare the `MeshBase` in namespace `libMesh`

Reply:
class MeshBase; ?
or class libMesh::MeshBase?

Reply:
both get me an error. first one is 'ambiguous' (two lines later), second says forward declarations cant be nested. 

maybe classMeshBase

then in the declaration
...... libMesh::MeshBase ....)
would work.

Reply:
```c++
namespace libMesh
{
class MeshBase;
}

void checkNonConformalMesh(const std::unique_ptr<libMesh::MeshBase> & mesh,
```

Reply:
done